### ISSUE TITLE
Allow characters to level up.

### ISSUE THREAD

**Issue by skillsreactor** [skillsreactor/rpg-learning-example#3]
We want to have some player character progression. We haven't done much math to deal with game balance at the moment, but don't necessarily want a linear leveling system. To resolve this issue, a leveling function will need to be proposed in order make each subsequent level-up take more experience.

## User Story

```
AS a player with a character
I WANT my character to grow stronger over time
SO THAT I can defeat more enemies between resting to retain my wealth and feel a sense of progression.
```

## Acceptance Criteria

```
GIVEN a created character
WHEN I defeat an enemy in combat
THEN I accumulate some experience
WHEN my accumulated experience exceeds a threshold
THEN my character level should be increased
WHEN my character is first created
THEN I should be level 1
WHEN my character level is increased
THEN my attack and health should be increased linearly
AND THEN my next level threshold should be increased in a non-linear way (based on a curve with current level as input)
```

## References

Here are some starting ideas (quick google search results)

https://www.davideaversa.it/blog/gamedesign-math-rpg-level-based-progression/
http://howtomakeanrpg.com/a/how-to-make-an-rpg-levels.html
https://onlyagame.typepad.com/only_a_game/2006/08/mathematics_of_.html

**Comment by Maniac-i** [comment]
my first iteration of this establishes a level, xp, and nextLevelAt attributes, and two methods, gainXP and levelUp.

xp is determined by selecting a random number between 10 - 30 (to match the enemies health possibilities) and multiplying it by 1 + (the characters level / 10). This multiplier maxes out at 2 (level 20).

XP the nextLevelAt is determined by the current xp need for the next level + (the current xp needed for the next level * 2) - (the current characters level * 50).

**Comment by skillsreactor** [comment]
Closed by #5
